src/app/shared/date-service/date-service.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DateService {
+  getCurrentDate() {
+    return new Date().toLocaleDateString();
+  }
+
+  getDate() {
+    return new Date();
+  }
+
+  compareDate(a, b) {
+    // 1, a > б
+    // 0, a = b
+    // -1, a < б
+
+    const left: any = a > b;
+    const right: any = a < b;
+    return isFinite((a = this.convert(a).valueOf())) &&
+      isFinite((b = this.convert(b).valueOf()))
+      ? left - right
+      : NaN;
+  }
```

Comment:
Looks like not needed overengineering or copy paste. If this code really needed, plz make sure it is covered with unit tests.